htdocs/fourn/facture/card.php

```diff
@@ -238,8 +238,11 @@
 	// Set supplier ref
 	if ($action == 'setref_supplier' && $user->rights->fournisseur->commande->creer)
 	{
-	    $result=$object->setValueFrom('ref_supplier',GETPOST('ref_supplier','alpha'));
-	    if ($result < 0) dol_print_error($db, $object->error);
+		$object->ref_supplier = GETPOST('ref_supplier', 'alpha');
+		
+		if ($object->update() < 0) {
+			setEventMessage($object->error, 'errors');
```

Comment:
Can you replace this with 
setEventMessages($object->error, $object->errors, 'errors')
so error introduced into the update will be correctly reported.